### PR TITLE
add atom: protocol

### PR DESCRIPTION
**Pull request by UziTech**
### Description of the Change

open `atom://` links

### Benefits

open files in atom

### Possible Drawbacks

none

### Applicable Issues

requires https://github.com/atom/language-hyperlink/pull/21

### CODE REVIEW

**Comment by UziTech** [comment]
@50Wliu this PR will work after https://github.com/atom/language-hyperlink/pull/21 is merged. And again neither of these PRs require https://github.com/atom/atom/pull/16940 to be merged.

**Comment by 50Wliu** [comment]
The tests for this need to be gated, probably >= Atom 1.32.

**Comment by UziTech** [comment]
How do we do that before 1.32.0 is released?

**Comment by 50Wliu** [comment]
Guard the test with something like `if(atom.getVersion().parseFloat() >= 1.32)` (pseudocode).

**Comment by 50Wliu** [comment]
One last thing: can you confirm this works when you link `language-hyperlink` locally?

File: spec/link-spec.js [hunk 1]
```diff
@@ -76,8 +112,8 @@ you should not [click][her]
       })
     )
 
-    it('does not open non http/https links', async () => {
-      await atom.workspace.open('sample.txt')
+    it('does not open non http/https/atom links', async () => {
+      await atom.workspace.open('sample.md')
```
Comment (UziTech) [inline spec/link-spec.js]: I updated the tests to use .md files since `language-gfm` is loaded and `language-text` is not

**Comment by UziTech** [comment]
I can confirm all 4 tests pass when the current master branch of `language-hyperlink` is loaded

**Comment by daviwil** [comment]
Thanks a lot @UziTech!

**Pull request merged**